server/src/business-layer/services/StripeService.ts

```diff
@@ -271,22 +271,26 @@ export default class StripeService {
       productUpdateParams
     )
 
-    /** Return the updated product */
+    /** Return the updated product
+     *
+     * @returns
+     */
     return updatedProduct
   }
 
+  /** Fetch all active products from Stripe
+   * @returns membershipProducts - An array of active membership products from Stripe
+   */
   public async getActiveMembershipProducts() {
     try {
-      // Fetch all active products from Stripe
       const products = await stripe.products.list({
         active: true
       })
       // Filter products with the required metadata
       const membershipProducts = products.data.filter(
         (product) =>
           product.metadata[MEMBERSHIP_TYPE_KEY] ===
-            MembershipTypeValues.NewNonStudent &&
-          product.metadata.discount === "true"
+          MembershipTypeValues.NonUoaStudent
```

Comment:
For this line, I was a bit confused. I assume it has to cover all MembershipTypeValues, but I wasn't sure how to implement this.

Reply:
Looks like this is the old version, you should be using the new keys you made in `StripeProductMetadata`

Reply:
You can look at some of the older commits to see what it should look like

Reply:
Oh yeah mb, I think I need to get some sleep lmao